src/Pinch/Internal/Value.hs

```diff
@@ -191,6 +194,7 @@ instance Hashable (Value a) where
       VInt64  x -> s `hashWithSalt` (6 :: Int) `hashWithSalt` x
       VList   x -> s `hashWithSalt` (7 :: Int) `hashWithSalt` x
       VMap    x -> s `hashWithSalt` (8 :: Int) `hashWithSalt` x
+      VNullMap  -> s `hashWithSalt` (8 :: Int)
```

Comment:
We should use a different salt value for each constructor. I would make this 11.